File: src/main/java/javabot/operations/time/TimezonesGreenland.java
package javabot.operations.time;

public class TimezonesGreenland implements Timezones {

    private static Tri<String> timezones = new Tri<String>( );

    static {
        for( Timezones timezone: Timezones.values( ) ) {
            timezones.insert( timezone.name( ), timezone.getTimezone( ) );
        }
    }

    private enum Timezones {
        Thule( "America/Thule" ),
        Godthab( "America/Godthab" ),
        Scoresbysund( "America/Scoresbysund" ),
        Danmarkshavn( "America/Danmarkshavn" );

        private String timezone;

        Timezones( String timezone ) {
            this.timezone = timezone;
        }

        public String getTimezone( ) {
            return timezone;
        }
    }

    @Override
    public String get( String province ) {
        return timezones.get( province );
    }

    @Override
    public String getCapital( ) {
        return Timezones.Godthab.getTimezone( );
    }
}

File: src/main/kotlin/javabot/operations/time/TimezonesGreenland.kt
package javabot.operations.time

public class TimezonesGreenland : Timezones {

    private enum class Timezones(public val timezone: String) {
        Thule("America/Thule"),
        Godthab("America/Godthab"),
        Scoresbysund("America/Scoresbysund"),
        Danmarkshavn("America/Danmarkshavn")
    }

    override fun get(province: String): String? {
        return timezones.get(province)
    }

    override fun getCapital(): String {
        return Timezones.Godthab.timezone
    }

    companion object {

        private val timezones = Tri<String>()

        init {
            for (timezone in Timezones.values()) {
                timezones.insert(timezone.name, timezone.timezone)
            }
        }
    }
}
